src/xclient_zmq_impl.cpp

```diff
@@ -166,14 +177,20 @@ namespace xeus
     void xclient_zmq_impl::start()
     {
         start_iopub_thread();
-        // TODO : Introduce a client, xheartbeat_client that runs on its own thread, m_heartbeat_thread.
+        // pass in desired timeout value below
+        start_heartbeat_thread();
```

Comment:
I think we can pass in a desired timeout value here as the `wait_for_answer` function would preferrably need a timeout (https://github.com/jupyter-xeus/xeus-zmq/blob/804088338c07240a8a32b830acc2d5537ee8f131/src/xheartbeat_client.cpp#L53)

Reply:
Also not sure what sort of timeout value are we looking at here.